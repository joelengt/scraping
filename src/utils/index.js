export {noop} from './noop'
export {validateParam, validateBody, schemas} from './validator'
