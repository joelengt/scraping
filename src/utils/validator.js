var debug = require('debug')('riqra-service-partner:utils-validator')
const Joi = require('joi')

// Validate uri param
export const validateParam = (schema) => {
  return (req, res, next) => {
    const result = Joi.validate(req.params, schema)
    if (result.error) {
      return res['400']({success: false, validator: result.error}, 'Los campos nos son validos')
    }

    if (!req.value) {
      req.value = {}
    }

    if (!req.value['params']) {
      req.value['params'] = {}
    }

    req.params = result.value
    next()
  }
}

// Validate body attributes
export const validateBody = (schemas) => {
  return (req, res, next) => {
    const result = Joi.validate(req.body, schemas)
    if (result.error) {
      return res['400']({success: false, validator: result.error}, 'Los campos nos son validos')
    }

    if (!req.value) {
      req.value = {}
    }

    if (!req.value['body']) {
      req.value['body'] = {}
    }

    req.body = result.value
    next()
  }
}

// schemas
export const schemas = {
  partnerIDSchema: Joi.object().keys({
    id: Joi.number().required()
  }),
  adsIDSchema: Joi.object().keys({
    id: Joi.number().required(),
    partnerId: Joi.number().required()
  }),
  partIDSchema: Joi.object().keys({
    partnerId: Joi.number().required()
  }),
  partnerSlugifySchema: Joi.object().keys({
    partnerSlugify: Joi.string().required()
  }),
  partnerSchema: Joi.object().keys({
    name: Joi.string().required(),
    name_slugify: Joi.string().required(),
    logo: Joi.string().required(),
    background_color: Joi.string(),
    product_add_button_color: Joi.string(),
    cart_button_color: Joi.string(),
    product_info_color: Joi.string(),
    product_arrows_color: Joi.string(),
    minimum_purchase: Joi.string(),
    delivery_message: Joi.string(),
    is_featured: Joi.boolean().strict(),
    is_guest_enabled_to_buy: Joi.boolean().strict()
  }),
  partnerOptionalSchema: Joi.object().keys({
    name: Joi.string(),
    name_slugify: Joi.string(),
    logo: Joi.string(),
    background_color: Joi.string(),
    product_add_button_color: Joi.string(),
    cart_button_color: Joi.string(),
    product_info_color: Joi.string(),
    product_arrows_color: Joi.string(),
    minimum_purchase: Joi.string(),
    delivery_message: Joi.string(),
    is_featured: Joi.boolean().strict(),
    is_guest_enabled_to_buy: Joi.boolean().strict()
  }),
  adsSchema: Joi.object().keys({
    name: Joi.string().required(),
    photo: Joi.string().required(),
    link: Joi.string().required()
  }),
  adsUpdatedSchema: Joi.object().keys({
    name: Joi.string().required(),
    photo: Joi.string().required(),
    link: Joi.string().required()
  }),
  adsOptionalSchema: Joi.object().keys({
    name: Joi.string(),
    photo: Joi.string(),
    link: Joi.string()
  })
}
