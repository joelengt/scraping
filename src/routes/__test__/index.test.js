// Test Rotues api
import {noop} from '../../utils'
require('dotenv').config()
var sql = require('../../initializers/knex')
var axios = require('axios')

var service = axios.create({
  headers: {'X-Requested-With': 'XMLHttpRequest'}
})

let URI = 'http://' + process.env.HOST + ':' + process.env.PORT

// Testing Status Code
test('it route create new ads item - POST /api/ads - statusCode 201', async () => {
  let body = {
    name: 'pretty banner',
    link: '/something.png',
    photo: 'http://cdn.image./something.png'
  }

  let response = await service.post(`${URI}/api/ads`, body)
  expect(response.status).toBe(201)
})

test('it route ads item by id - GET /api/ads/:id - statusCode 200', async () => {
  let UltimateElement = await sql('banner')
  .orderBy('id', 'desc')
  .limit(1)
  .spread(noop)

  let response = await service.get(`${URI}/api/ads/${UltimateElement.id}`)
  expect(response.status).toBe(200)
})

test('it route ads items - GET /api/ads - statusCode 200', async () => {
  let response = await service.get(`${URI}/api/ads`)
  expect(response.status).toBe(200)
})

test('it route update ads item by id - PUT /api/ads/:id - statusCode 200', async () => {
  let body = {
    name: 'pretty banner',
    link: '/something.png',
    photo: 'http://cdn.image./something.png'
  }

  let UltimateElement = await sql('banner')
  .orderBy('id', 'desc')
  .limit(1)
  .spread(noop)

  let response = await service.post(`${URI}/api/ads/${UltimateElement.id}?_method=put`, body)
  expect(response.status).toBe(200)
})

test('it route delete ads item by id - DELETE /api/ads/:id - statusCode 200', async () => {
  let UltimateElement = await sql('banner')
  .orderBy('id', 'desc')
  .limit(1)
  .spread(noop)

  let response = await service.post(`${URI}/api/ads/${UltimateElement.id}?_method=delete`)
  expect(response.status).toBe(200)
})

// Testing response json
test('it route create new ads item - POST /api/ads', async () => {
  let body = {
    name: 'pretty banner',
    link: '/something.png',
    photo: 'http://cdn.image./something.png'
  }

  let response = await service.post(`${URI}/api/ads`, body)
  expect(response.data).toMatchSnapshot()
})

test('it route ads item by id - GET /api/ads/:id', async () => {
  // Get ultimes item created
  let UltimateElement = await sql('banner')
  .orderBy('id', 'desc')
  .limit(1)
  .spread(noop)

  let response = await service.get(`${URI}/api/ads/${UltimateElement.id}`)
  expect(response.data).toMatchSnapshot()
})

test('it route ads items - GET /api/ads', async () => {
  let response = await service.get(`${URI}/api/ads`)
  expect(response.data).toMatchSnapshot()
})

test('it route update ads item by id - PUT /api/ads/:id', async () => {
  let body = {
    name: 'pretty banner',
    link: '/something.png',
    photo: 'http://cdn.image./something.png'
  }

  let UltimateElement = await sql('banner')
  .orderBy('id', 'desc')
  .limit(1)
  .spread(noop)

  let response = await service.post(`${URI}/api/ads/${UltimateElement.id}?_method=put`, body)
  expect(response.data).toMatchSnapshot()
})

test('it route delete ads item by id - DELETE /api/ads/:id', async () => {
  let UltimateElement = await sql('banner')
  .orderBy('id', 'desc')
  .limit(1)
  .spread(noop)

  let response = await service.post(`${URI}/api/ads/${UltimateElement.id}?_method=delete`)
  expect(response.data).toMatchSnapshot()
})
