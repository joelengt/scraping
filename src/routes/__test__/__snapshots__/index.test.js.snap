// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it route ads item by id - GET /api/ads/:id 1`] = `
Object {
  "data": Object {
    "item": Object {
      "created_at": "2017-06-14T08:23:15.000Z",
      "id": 11,
      "link": "/something.png",
      "name": "pretty banner",
      "photo": "http://cdn.image./something.png",
      "updated_at": "2017-06-14T08:23:15.000Z",
    },
  },
  "message": "Elemento Ads encontrado",
  "status": 200,
}
`;

exports[`it route ads items - GET /api/ads 1`] = `
Object {
  "data": Object {
    "ads": Array [
      Object {
        "created_at": "2017-05-22T23:45:43.000Z",
        "id": 4,
        "link": "/ofertas",
        "name": "Ofertas",
        "photo": "https://res.cloudinary.com/riqra/image/upload/v1487711086/web/offers-banner.jpg",
        "updated_at": "2017-05-22T23:45:43.000Z",
      },
      Object {
        "created_at": "2017-06-13T21:46:50.000Z",
        "id": 6,
        "link": "http://cnd.image/awesome.png",
        "name": "Pretty banner",
        "photo": "awesome.png",
        "updated_at": "2017-06-13T21:46:50.000Z",
      },
      Object {
        "created_at": "2017-06-13T21:46:52.000Z",
        "id": 7,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-13T21:46:52.000Z",
      },
      Object {
        "created_at": "2017-06-13T22:09:22.000Z",
        "id": 9,
        "link": "http://cdn.image./something.png1",
        "name": "pretty banner1",
        "photo": "something.png1",
        "updated_at": "2017-06-13T22:09:22.000Z",
      },
      Object {
        "created_at": "2017-06-14T08:23:15.000Z",
        "id": 11,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T08:23:15.000Z",
      },
      Object {
        "created_at": "2017-06-14T16:24:51.000Z",
        "id": 12,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T16:24:51.000Z",
      },
      Object {
        "created_at": "2017-06-14T16:27:03.000Z",
        "id": 15,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T16:27:03.000Z",
      },
      Object {
        "created_at": "2017-06-14T16:28:28.000Z",
        "id": 16,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T16:28:28.000Z",
      },
      Object {
        "created_at": "2017-06-14T16:29:42.000Z",
        "id": 17,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T16:29:42.000Z",
      },
      Object {
        "created_at": "2017-06-14T16:29:51.000Z",
        "id": 18,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T16:29:51.000Z",
      },
      Object {
        "created_at": "2017-06-14T17:06:12.000Z",
        "id": 20,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T17:06:12.000Z",
      },
      Object {
        "created_at": "2017-06-14T17:07:58.000Z",
        "id": 21,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T17:07:58.000Z",
      },
      Object {
        "created_at": "2017-06-14T17:08:14.000Z",
        "id": 22,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T17:08:14.000Z",
      },
      Object {
        "created_at": "2017-06-14T17:08:17.000Z",
        "id": 23,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T17:08:17.000Z",
      },
      Object {
        "created_at": "2017-06-14T17:31:10.000Z",
        "id": 24,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T17:31:10.000Z",
      },
      Object {
        "created_at": "2017-06-14T17:31:12.000Z",
        "id": 25,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T17:31:12.000Z",
      },
      Object {
        "created_at": "2017-06-14T17:42:10.000Z",
        "id": 26,
        "link": "http://cdn.image./something.png",
        "name": "pretty banner",
        "photo": "something.png",
        "updated_at": "2017-06-14T17:42:10.000Z",
      },
      Object {
        "created_at": "2017-06-14T19:46:44.000Z",
        "id": 27,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T19:46:44.000Z",
      },
      Object {
        "created_at": "2017-06-14T19:47:15.000Z",
        "id": 28,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T19:47:15.000Z",
      },
      Object {
        "created_at": "2017-06-14T19:55:30.000Z",
        "id": 29,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T19:55:30.000Z",
      },
      Object {
        "created_at": "2017-06-14T19:56:40.000Z",
        "id": 30,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T19:56:40.000Z",
      },
      Object {
        "created_at": "2017-06-14T19:57:02.000Z",
        "id": 31,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T19:57:02.000Z",
      },
      Object {
        "created_at": "2017-06-14T19:57:32.000Z",
        "id": 32,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T19:57:32.000Z",
      },
      Object {
        "created_at": "2017-06-14T19:59:57.000Z",
        "id": 33,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T19:59:57.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:00:35.000Z",
        "id": 34,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:00:35.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:01:51.000Z",
        "id": 36,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:01:51.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:16:12.000Z",
        "id": 37,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:16:12.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:17:35.000Z",
        "id": 38,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:17:35.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:18:12.000Z",
        "id": 39,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:18:12.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:18:32.000Z",
        "id": 40,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:18:32.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:20:25.000Z",
        "id": 41,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:20:25.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:21:20.000Z",
        "id": 42,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:21:20.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:27:10.000Z",
        "id": 43,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:27:10.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:29:04.000Z",
        "id": 44,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:29:04.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:29:26.000Z",
        "id": 45,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:29:26.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:29:40.000Z",
        "id": 46,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:29:40.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:30:52.000Z",
        "id": 47,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:30:52.000Z",
      },
      Object {
        "created_at": "2017-06-14T20:31:19.000Z",
        "id": 48,
        "link": "/something.png",
        "name": "pretty banner",
        "photo": "http://cdn.image./something.png",
        "updated_at": "2017-06-14T20:31:19.000Z",
      },
    ],
  },
  "message": "Producto actualizado en el carrito",
  "status": 200,
}
`;

exports[`it route update ads item by id - PUT /api/ads/:id 1`] = `
Object {
  "data": Object {
    "id": Object {
      "created_at": "2017-06-14T08:23:15.000Z",
      "id": 11,
      "link": "/something.png",
      "name": "pretty banner",
      "photo": "http://cdn.image./something.png",
      "updated_at": "2017-06-14T08:23:15.000Z",
    },
  },
  "message": "Elemento Ads actualizado",
  "status": 200,
}
`;
